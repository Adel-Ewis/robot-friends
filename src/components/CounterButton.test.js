import { shallow } from 'enzyme';
import CounterButton from './CounterButton';
import React from 'react';

it('carList renders', () => {
  const mockColor = 'red';
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it('correctly increments the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.instance().props.color).toEqual('red');
});
