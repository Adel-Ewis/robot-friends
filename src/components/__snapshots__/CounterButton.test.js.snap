// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carList renders 1`] = `
<div>
  <button
    color="red"
    id="counter"
    onClick={[Function]}
  >
    Count: 
    0
  </button>
</div>
`;
